playbooks/freeipa/tasks/main.yaml

```diff
@@ -0,0 +1,62 @@
+---
+- name: 'Get Katello Hostname'
+  shell: hostname -f
+  delegate_to: "{{ katello_server }}"
+  register: katello_fqdn
+
+- name: 'Get FreeIPA Shortname'
+  shell: hostname -s
+  register: freeipa_shortname
+
+- name: 'Install FreeIPA server package (this could take a while)'
+  yum: name=ipa-server state=latest
+
+- name: 'Install FreeIPA DNS Package'
+  yum: name=ipa-server-dns state=latest
+
+- name: 'Install Haveged (faster population of PRNG)'
+  yum: name=haveged state=latest
+
+- name: 'Start Haveged'
+  service: name=haveged state=started
+
+- name: 'Fix /etc/hosts record for self'
+  lineinfile: dest=/etc/hosts regexp=".*{{ ansible_nodename }}.*localhost.*" line="{{ freeipa_ip }} {{ ansible_nodename }}" state=present
```

Comment:
Newer vegarants create a host record like this:

`127.0.0.1 myhost myhost.example.com localhost.domain localhost`

Which is totally incompatbile with freeipa, it complains about the fqdn resolving to 127.0.0.1. So this line, replaces that with a `192.168.121.2 myhost.example.com myhost` and localhost only is restored below.